sdk/vision/azure-ai-vision-imageanalysis/src/test/java/com/azure/ai/vision/imageanalysis/ImageAnalysisAsyncClientTest.java

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.vision.imageanalysis.tests;
+
+import java.util.Arrays;
+import java.util.List;
+import java.net.MalformedURLException;
+
+import org.junit.jupiter.api.Test;
+
+import com.azure.ai.vision.imageanalysis.*;
+import com.azure.ai.vision.imageanalysis.models.*;
+
+class ImageAnalysisAsyncClientTest extends ImageAnalysisClientTestBase {
+
+    private final Boolean sync = false; // All tests in this file use the async client
+
+    /***********************************************************************************
+     *
+     *                            HAPPY PATH TESTS
+     *
+     ***********************************************************************************/
+
+    @Test
+    public void testAnalyzeAsyncAllFeaturesFromFile() throws MalformedURLException {
+
+        createClientForStandardAnalysis(sync);
+
+        String methodName = new Object(){}.getClass().getEnclosingMethod().getName();
+        String imageSource = imageFile;
+        List<VisualFeatures> visualFeatures =  Arrays.asList(
+                VisualFeatures.SMART_CROPS,
+                VisualFeatures.CAPTION,
+                VisualFeatures.DENSE_CAPTIONS,
+                VisualFeatures.OBJECTS,
+                VisualFeatures.PEOPLE,
+                VisualFeatures.READ,
+                VisualFeatures.TAGS);
+        ImageAnalysisOptions options = null;
+        doAnalysis(methodName, sync, imageSource, visualFeatures, options);
+    }
+
+    @Test
+    public void testAnalyzeAsyncSingleFeatureFromUrl() throws MalformedURLException {
+
+        createClientForStandardAnalysis(sync);
+
+        String methodName = new Object(){}.getClass().getEnclosingMethod().getName();
+        String imageSource = imageUrl;
+
+        List<VisualFeatures> visualFeatures =  Arrays.asList(VisualFeatures.DENSE_CAPTIONS);
+        ImageAnalysisOptions options = new ImageAnalysisOptionsBuilder().setGenderNeutralCaption(true).build();
+        doAnalysis(methodName + ":DenseCaptions", sync, imageSource, visualFeatures, options);
+
+        visualFeatures =  Arrays.asList(VisualFeatures.SMART_CROPS);
+        options = new ImageAnalysisOptionsBuilder().setSmartCropsAspectRatios(Arrays.asList(0.9, 1.33)).build();
+        doAnalysis(methodName + ":SmartCrops", sync, imageSource, visualFeatures, options);
+
+        visualFeatures =  Arrays.asList(VisualFeatures.TAGS);
+        options = new ImageAnalysisOptionsBuilder().setLanguage("en").build();
+        doAnalysis(methodName + ":Tags", sync, imageSource, visualFeatures, options);
+
+        visualFeatures =  Arrays.asList(VisualFeatures.PEOPLE);
+        doAnalysis(methodName + ":People", sync, imageSource, visualFeatures, null);
```

Comment:
I'd recommend turning this into multiple tests, if the first one fails we don't know if the latter ones fail and may need to iterate many times before the test passes.

Reply:
@alzimmermsft thank you for this comment. Some tests do a single Image Analysis call, other tests (like the ones you pointed out) do multiple Image Analysis calls using the same ImageAnalysisClient. It's important that we have a test or two that do multiple calls using the same client, as this will be the common use case (creating a single client and using it to analyze, for example, a folder full of images). So, I'd rather keep this test as-is if that's okay with you. Reach out to me over IM if you want to discuss this further.